payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -157,72 +165,72 @@ internal class PaymentSheetViewModel internal constructor(
+                stripeIntentValidator.requireValid(stripeIntent)
             }.fold(
                 onSuccess = {
-                    _paymentIntent.value = paymentIntent
-                    resetViewState(paymentIntent, userErrorMessage = null)
+                    _stripeIntent.value = stripeIntent
+                    resetViewState(stripeIntent, userErrorMessage = null)
                 },
                 onFailure = ::onFatal
             )
         }
     }
 
     /**
-     * There's nothing left to be done in payment sheet if the PaymentIntent is confirmed.
+     * There's nothing left to be done in payment sheet if the [StripeIntent] is confirmed.
      *
      * See [How intents work](https://stripe.com/docs/payments/intents) for more details.
      */
-    private fun onConfirmedPaymentIntent(paymentIntent: PaymentIntent) {
+    private fun onConfirmedStripeIntent(stripeIntent: StripeIntent) {
         logger.info(
             """
-            PaymentIntent with id=${paymentIntent.id}" has already been confirmed.
+            ${stripeIntent.javaClass.simpleName} with id=${stripeIntent.id} has already been confirmed.
             """.trimIndent()
         )
         _viewState.value = PaymentSheetViewState.FinishProcessing {
-            processResult(
-                PaymentIntentResult(
-                    paymentIntent,
-                    StripeIntentResult.Outcome.SUCCEEDED
-                )
-            )
+            _paymentSheetResult.value = PaymentSheetResult.Completed
         }
     }
 
-    private fun resetViewState(paymentIntent: PaymentIntent, @IntegerRes stringResId: Int?) {
+    private fun resetViewState(stripeIntent: StripeIntent, @IntegerRes stringResId: Int?) {
         resetViewState(
-            paymentIntent,
+            stripeIntent,
             stringResId?.let { getApplication<Application>().resources.getString(it) }
         )
     }
 
-    private fun resetViewState(paymentIntent: PaymentIntent, userErrorMessage: String?) {
-        val amount = paymentIntent.amount
-        val currencyCode = paymentIntent.currency
-        _processing.value = false
-        if (amount != null && currencyCode != null) {
-            _amount.value = Amount(amount, currencyCode)
+    private fun resetViewState(stripeIntent: StripeIntent, userErrorMessage: String?) {
+        if (stripeIntent is PaymentIntent) {
+            val amount = stripeIntent.amount
+            val currencyCode = stripeIntent.currency
+            if (amount != null && currencyCode != null) {
+                _amount.value = Amount(amount, currencyCode)
+                _viewState.value =
+                    PaymentSheetViewState.Reset(userErrorMessage?.let { UserErrorMessage(it) })
+            } else {
+                onFatal(
+                    IllegalStateException("PaymentIntent could not be parsed correctly.")
+                )
+            }
+        } else {
```

Comment:
feels like worthwhile logging something like "incorrect state, resetting with StripeIntent while only PaymentIntent is allowed" here